### PR TITLE
adding solution to task8

### PR DESCRIPTION
**Pull request by sharontakhar**
hello, please can i submit by solution to task 8 please,

i have names the functions getArticles/selectArticles

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -240,3 +240,47 @@ describe("4.GET /api/users", () => {
...
+        });
+      });
+  });
+
+  test("the first array to be in DESC order to be equal to the below", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.articles[0]).toEqual(
```
Comment (alexdevdavis) [inline __tests__/index.test.js]: this test doesn't test the order of the res.body array: you're going to need jest sorted npm package to get this done efficiently

File: models/models.js [hunk 2]
```diff
@@ -29,6 +27,17 @@ exports.selectArticles = (article_id) => {
     });
 };
 
+//GET API ARTICLES
+exports.selectArticles = () => {
+  return db
+    .query(
+      "SELECT articles.*, CAST(COUNT(comments.article_id)AS INT) AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY articles.created_at ASC"
```
Comment (alexdevdavis) [inline models/models.js]: your db query should change as a result of arguments passed into this function (drawn from req.params)
you'll also need to use <array variable>.includes to protect against SQL injection.

I'd suggest having a look through the notes for this one

**Review by alexdevdavis — CHANGES_REQUESTED** [review]
as per comments, your query should group and/or sort by whatever variables are passed into the function

**Comment by k-huggs** [comment]
merge!

**Pull request merged**